### ISSUE TITLE
How might one set the starting scroll position?

### ISSUE THREAD

**Issue by patsplat** [Norserium/react-indiana-drag-scroll#4]
Is this a feature request?  Or am I missing something obvious?

Thanks!

**Comment by Norserium** [comment]
Yes and no.

Me need to consider, what's the best way to solve this problem: add the `scrollTo` method to the `ScrollContainer` or add a method to get its root DOM element.

Because, technically, you just need to scroll the root element of the `ScrollContainer` component. Anyway, currently you can get it himself by using `ReactDOM.findDOMNode()`.
https://codesandbox.io/s/react-example-816ln

It looks like workaround, but it works now, while your feature request will be considering.

**Comment by patsplat** [comment]
Yes, I was able to make it work with a ref to the container.

Also encountered a race condition on startup.

On my outdated Chromium (don't ask) the react js was running before the
element could actually scroll.  Fixed that with a pair of methods, one
which would scroll and another which would check that the scroll was
successfully 100ms later.

**Comment by Norserium** [comment]
Are you sure that it's the problem of the current library? Could you test it at a simple scrollable container? Can you reproduce this bug at my [example](https://codesandbox.io/s/react-example-816ln)?

**Comment by patsplat** [comment]
Oh the bug was definitely was due to my device, not the library.  Only
mentioned it and the workaround for others who might find this thread.

The library is great, thanks!

**Comment by Norserium** [comment]
Anyway, I will consider the mentioned features, and during this period I leave your issue open. It could be useful for other users.

Good luck!

**Comment by k1sul1** [comment]
It definitely was useful for me! I want to "remember" the scroll position so it can be restored when user comes back. 

I had to wrap the scrollTo call in a setTimeout, otherwise it doesn't work on subsequent renders. I can't avoid the re-renders because when the user clicks on a link in the Timeline component, react-router will display an additional route, and after they hit back, react-router renders again.

I'm using this component only once in the entire app, so this dirty solution works for me: 

```javascript
let scrollMemoryStore = { scrollLeft: 0, scrollTop: 0 }

class Timeline extends Component {
  container = React.createRef()

  onScrollEnd = (scrollLeft, scrollTop, scrollWidth, scrollHeight) => {
    scrollMemoryStore = { scrollLeft, scrollTop }
  }

  componentDidMount() {
    const node = ReactDOM.findDOMNode(this.container.current)

    setTimeout(() => {
      node.scrollTo(scrollMemoryStore.scrollLeft, scrollMemoryStore.scrollTop)
    })
  }

  render() {
    return <ScrollContainer ref={this.container} onEndScroll={this.onScrollEnd} />
  }
}
```

**Comment by Norserium** [comment]
Frankly, I don't suppose that this solution is so dirty.

But, anyway, do you suppose that using of `setTimeout` is caused by this library? Could you test your code at simple scrollable block?

**Comment by k1sul1** [comment]
It's a bit dirty because it's simply broken if you have two instances of the component, with different data. Easily fixed with a key though. I could plug it into redux, but nah.

I think I've to use setTimeout because my render is rather heavy; Timeline consists of over 30 entries, and each entry has a html parsing component inside.

**Comment by Norserium** [comment]
I see. But I assume, that the problem with saving of scroll position is not related to the current library, it's a more general problem.

**Comment by k1sul1** [comment]
The way I see it, your library works as it should. 

My usecase and way of doing things aren't exactly standard. I could make my redux state aware of the ongoing html renders, but then I'd have even more re-renders so I think working around it is the only worthwhile solution.

**Comment by k1sul1** [comment]
I found out that the reason I had to use setTimeout is because I used React.lazy to load the component containing the scroll container, and the component was being unmounted and remounted on every prop change to the parent. I cached the results of React.lazy and now the component isn't doing unnecessary renders. I don't need the fade in animation anymore either.

Heaviness of the render doesn't seem to have any effect on the scroll restoration, like I thought at first.
